src/hasher512.cpp

```diff
@@ -160,9 +160,9 @@ namespace libtorrent
 #endif
 		}
 #elif defined TORRENT_USE_LIBCRYPTO
-		SHA512_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data()), data.size());
+		SHA512_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data()), std::uint32_t(data.size()));
```

Comment:
libTorrent SHA512_Update & SHA1_update routines use std::uint32_t as length arguments. You think it's better to transfer them to std::size_t ?